schemas/github-pull-request-message.yml

```diff
@@ -0,0 +1,39 @@
+id: http://schemas.taskcluster.net/github/v1/github-pull-request-message.json#
+$schema:  http://json-schema.org/draft-04/schema#
+title:        "GitHub Pull Request Message"
+description: |
+  Message reporting that a GitHub pull request has occurred
+type:         object
+properties:
+  version:    {$const: message-version}
+  organizationName:
+    description: |
+      The GitHub `organizationName` which had an event.
+    type:         string
+    minLength:    {$const: identifier-min-length}
+    maxLength:    {$const: identifier-max-length}
+    pattern:      {$const: identifier-pattern}
+  repositoryName:
+    description: |
+      The GitHub `repositoryName` which had an event.
+    type:         string
+    minLength:    {$const: identifier-min-length}
+    maxLength:    {$const: identifier-max-length}
+    pattern:      {$const: identifier-pattern}
+  action:
+    description: |
+      The GitHub `action` which triggered an event.
+    type:         string
+    minLength:    {$const: identifier-min-length}
+    maxLength:    {$const: identifier-max-length}
+    pattern:      {$const: identifier-pattern}
+  details:
+    type:         object
+    description: |
+      Metadata describing the pull request.
```

Comment:
I have mixed feeling about this... It's much easier to ensure correctness if we know what is present.
Also, we might not want to send giant messages, small pulse messages are great. So perhaps we should just include who pushed it, who committed the head, revision, things that we are definitely going to use and then add things as we need them.

I know it was a problem with the old hg publisher that it listed all files changed.. which made it unreliable because messages were too big.

Also if github APIs were to change, it would be nice, if we can just keep our stuff isolated. So even if you decide to just outline and include most of the details from the github payload, I still think a schema for all properties has huge value.